sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py

```diff
@@ -0,0 +1,1185 @@
+
+__author__ = ["angus924", "michaelfeil"]
+__all__ = ["MiniRocketMultivariateVariable"]
+
+import multiprocessing
+import warnings
+from typing import List, Union
+
+import numpy as np
+import pandas as pd
+from numba import get_num_threads, njit, prange, set_num_threads, vectorize
+
+from sktime.transformations.base import BaseTransformer
+
+
+class MiniRocketMultivariateVariable(BaseTransformer):
+    """MINIROCKET (Multivariate, unequal length).
+
+    MINImally RandOm Convolutional KErnel Transform. [1]_
+
+    **Multivariate**
+    **unequal length**
+
+    A provisional and naive extension of MINIROCKET to multivariate input
+    with unequal length provided by the authors [2]_ .  For better
+    performance, use the sktime class MiniRocket for univariate input,
+    and MiniRocketMultivariate to equal length multivariate input.
+
+    Parameters
+    ----------
+    num_kernels              : int, number of random convolutional kernels
+                               (default 10_000).
+                               Reduced to the closest multiple of 84.
+    max_dilations_per_kernel : int, maximum number of dilations per kernel (default 32)
+    reference_length         : int or str, length of reference, (default `'max'`), str
+                               options are `'max'`, `'mean'`, `'median'`, `'min'`
+    n_jobs                   : int, optional (default=1) The number of jobs to run in
+    parallel for `transform`. ``-1`` means using all processors.
+    pad_value_short_series   : float or None, padding series with len<9 to value
+                                    (default None)
+    random_state             : int, random seed (optional, default None)
+
+    Examples
+    --------
+    >>> from sktime.transformations.panel.rocket import MiniRocketMultivariateVariable
+    >>> from sktime.datasets import load_japanese_vowels
+    >>> # load multivariate and unequal length dataset
+    >>> X_train, _ = load_japanese_vowels(split="train", return_X_y=True)
+    >>> X_test, _ = load_japanese_vowels(split="test", return_X_y=True)
+    >>> pre_clf = MiniRocketMultivariateVariable(pad_value_short_series=0.0)
+    >>> pre_clf.fit(X_train, y=None)
+    MiniRocketMultivariateVariable(...)
+    >>> X_transformed = pre_clf.transform(X_test)
+    >>> X_transformed.shape
+    (370, 9996)
+
+
+    References
+    ----------
+    .. [1] @article{dempster_etal_2020,
+      author  = {Dempster, Angus and Schmidt, Daniel F and Webb, Geoffrey I},
+      title   = {{MINIROCKET}: A Very Fast (Almost) Deterministic Transform for
+                 Time Series Classification},
+      year    = {2020},
+      journal = {arXiv:2012.08791}
+    }
+
+    .. [2] Angus Dempster, Daniel F Schmidt, Geoffrey I Webb
+           https://github.com/angus924/minirocket
+    """
+
+    _tags = {
+        "univariate-only": False,
+        "fit_is_empty": False,
+        "scitype:transform-input": "Panel",
+        "scitype:transform-output": "Panel",
```

Comment:
These should be consistent with the other transformers.

Reply:
indeed, the ` "scitype:transform-output": "Panel"`. Can be modified to to `primitive`, like in the other rockets.

`"scitype:transform-input"` can't be modified to numpy3d, as the input data is unequal length time series data (+multivariate). Therefore, this minirocket needs an extra class from sktime. 
MiniRocket / MiniRocketMultivariate strictly maybe do not, however, there is a significant amount of different numba code as well.

Reply:
These tags should not change the output at all, the mtype tag for df-list is what sets the input datatype.

It is slightly odd but current convention is to set the tags
```
        "scitype:transform-input": "Series",
        "scitype:transform-output": "Primitives",
```
for transformers which take time series as an input and output primitives (even if the input is a panel).

If I'm wrong and this does actually impact the transformer, let me know.

Reply:
Indeed, now it works!